developer_manual/digging_deeper/dashboard.rst

```diff
@@ -0,0 +1,150 @@
+=========
+Dashboard
+=========
+
+The dashboard app aims to provide the user with a general overview of their 
+Nextcloud and show information that is currently important.
+
+App developers can integrate into the dashboard app and provide their own panels.
+
+
+Register a dashboard panels
+===========================
+
+A dashboard panel is represented by a class implementing the `OCP\\Dashboard\\IPanel` 
+interface. This class will be instantiated whenever the dashboard is loaded, so 
+scripts and initial state that your panel needs can be added in the constructor.
+
+.. code-block:: php
+
+    <?php
+
+    namespace OCA\MyApp\Dashboard;
+
+    use OCP\Dashboard\IPanel;
+    use OCP\IInitialStateService;
+    use OCP\IL10N;
+    use OCP\IURLGenerator;
+    use OCP\IUserSession;
+
+    class MyAppPanel implements IPanel {
+
+        public function __construct(
+            IInitialStateService $initialStateService,
+            IL10N $l10n,
+            IURLGenerator $urlGenerator
+        ) {
+            $this->l10n = $l10n;
+            $this->urlGenerator = $urlGenerator;
+
+            $initialStateService->provideInitialState('myapp', 'myData', []);
+            \OCP\Util::addScript('myapp', 'dashboard');
+        }
+
+        public function getId(): string {
+            return 'myapppanelid';
+        }
+
+        public function getTitle(): string {
+            return $this->l10n->t('My app');
+        }
+
+        public function getOrder(): int {
+            return 0;
+        }
+
+        public function getIconClass(): string {
+            return 'icon-class';
+        }
+
+        public function getUrl(): string {
```

Comment:
what should this URL point to?

Reply:
I've added a description in phpdoc for each method now